mewarpx/examples/thermionic_diode_patchy_cathode.py

```diff
@@ -0,0 +1,195 @@
+        self.SAVE = SAVE
+        self.DELTA_PHI = DELTA_PHI
+        self.PATCH_SIZE = PATCH_SIZE
+        self.USE_SCHOTTKY = USE_SCHOTTKY
+
+        self.DIAG_STEPS = DIAG_STEPS
+        if self.DIAG_STEPS is None:
+            self.DIAG_STEPS = self.TOTAL_TIMESTEPS // 5
+
+    def setup_run(self):
+
+        ####################################
+        # Diode setup
+        ####################################
+
+        self.run = diode_setup.DiodeRun_V1(
+            GEOM_STR='XZ',
+            CATHODE_TEMP=self.CATHODE_TEMP,
+            CATHODE_A=self.CATHODE_A,
+            CATHODE_PHI=self.CATHODE_PHI,
+            USE_SCHOTTKY=self.USE_SCHOTTKY,
+            ANODE_TEMP=self.ANODE_TEMP,
+            ANODE_PHI=self.ANODE_PHI,
+            V_ANODE_CATHODE=self.V_ANODE_CATHODE,
+            D_CA=self.D_CA,
+            DT=self.DT,
+            NX=self.NX,
+            NZ=self.NZ,
+            DIRECT_SOLVER=False,
+            NPPC=self.NPPC,
+            TOTAL_TIMESTEPS=self.TOTAL_TIMESTEPS,
+            DIAG_STEPS=self.DIAG_STEPS,
+            FIELD_DIAG_PLOT=self.FIELD_DIAG_PLOT
+        )
+
+        self.run.init_base()
+        self.run.init_electrons()
+
+        ####################################
+        # Conductor setup
+        ####################################
+
+        self.patchy_cathode = assemblies.PatchyCathode(
+            V=self.run.V_CATHODE, T=self.run.CATHODE_TEMP,
+            phi_bar=self.run.CATHODE_PHI, delta_phi=self.DELTA_PHI,
+            patch_size=self.PATCH_SIZE
+        )
+        self.run.cathode = self.patchy_cathode.cathode_list
+        self.run.electrons.save_particles_at_eb = 1
+
+        self.run.anode = assemblies.Anode(
+            z=self.run.D_CA, V=self.run.V_ANODE, T=self.run.ANODE_TEMP,
+            WF=self.run.ANODE_PHI
+        )
+        self.run.electrons.save_particles_at_zhi = 1
+
+        self.run.surface_list = self.run.cathode + [self.run.anode]
+
+        self.run.init_solver()
+
+        ####################################
+        # Injector setup
+        ####################################
+
+        self.run.emitter = emission.ZPlanePatchyEmitter(
+            patchy_cathode=self.patchy_cathode,
+            ymin=0.0, ymax=0.0,
+            transverse_fac=self.run.TRANSVERSE_FAC
+        )
+
+        self.run.injector1 = emission.ThermionicInjector(
+            self.run.emitter.high_wf, self.run.electrons, self.run.NPPC,
+            A=self.run.CATHODE_A, use_Schottky=self.run.USE_SCHOTTKY
+        )
+        self.run.injector2 = emission.ThermionicInjector(
+            self.run.emitter.low_wf, self.run.electrons, self.run.NPPC,
+            A=self.run.CATHODE_A, use_Schottky=self.run.USE_SCHOTTKY
+        )
+
+        ####################################
+        # Runinfo
+        ####################################
+
+        injector_dict = {
+            'cathode': self.run.injector1,
+            'cathode2': self.run.injector2
```

Comment:
Using 'cathode2' is fine for now, but a reminder that if we ever calculate efficiency rather than just power density, we'll need to make sure all the cathode objects are grouped together and separated from anode objects.

Reply:
Thanks! It actually works fine to just use `'cathode': [self.run.injector1, self.run.injector2]`. However, the surface flux diagnostic does not allow compound surfaces.